src/hal/mod.rs

```diff
@@ -22,14 +22,23 @@ and each such struct has a `setup()` method that configures the hardware
 */
 
 
+#[cfg(feature = "mcu_lpc17xx")]
```

Comment:
That's actually a bad idea I guess, as it feature-gates mcu code from being unit tested.